index.d.ts

```diff
@@ -0,0 +1,12 @@
+import { Parser } from 'acorn' 
+
+declare const jsx: (options?: jsx.AcornJSXOptions) => (BaseParser: typeof Parser) => typeof Parser;
+
+declare namespace jsx {
+  interface AcornJSXOptions {
+    allowNamespacedObjects?: boolean;
+    allowNamespaces?: boolean;
+  }
+}
+
+export = jsx;
```

Comment:
It would be better to use proper ES6 export declarations where possible - in particular, there's no reason for `namespace jsx`, instead you can just do `export interface ...` directly.

Reply:
I totally agree. However, since the function is exported using `module.exports = /* … */`, the correct way to reflect this using TypeScript is to use `export = /* … */`. The namespace is needed to export the options interface. It could be removed, but that would make the interface private.

Reply:
Right, because you can't combine `export =` with any named exports, even typings. Sigh, that's unfortunate, but ok. In the future we should just switch to ESM for the JS part...

Reply:
OTOH, are there any use-cases where someone needs to import the interface? Normally they only pass an object literal.

Reply:
Perhaps someone would like to do something like:

```tsx
import { Options as AcornOptions, Parser } from 'acorn';
import * as jsx from 'acorn-jsx'

interface Options {
  jsx: jsx.AcornJSXOptions;
  acorn: AcornOptions;
}

export function parseJSX(code: string, options: Options) {
  return Parser.extend(jsx(options.jsx)).parse(code, options.jsx);
}
```

From what I’ve seen, most packages export their options as a type.